File: generators/HttpResponseHeaderCodeGenerator/Templates/root.template.kt
package your.package

/**
 * HTTP Response Header
 */
enum class HttpResponseHeader(val value: String) {

%%ITEMS%%
    ;


    companion object {

        /**
         * Returns the element matching the given value.
         */
        @JvmStatic
        fun parse(value: String): HttpResponseHeader {
            return values().single { it.value == value }
        }

        /**
         * Returns the element matching the given value or null.
         */
        @JvmStatic
        fun parseOrNull(value: String): HttpResponseHeader? {
            return values().singleOrNull { it.value == value }
        }
    }
}

File: generators/HttpResponseHeaderCodeGenerator/Templates/root.template.swift
/**
 Http Response Header
 */
enum HttpResponseHeader: String {

    /// Returns the element matching the given value.
    public static func parse(value: String) -> HttpResponseHeader {
        return HttpResponseHeader.init(rawValue: value)!
    }

    /// Returns the element matching the given value or null.
    public static func parseOrNull(value: String) -> HttpResponseHeader? {
        return HttpResponseHeader.init(rawValue: value)
    }


%%ITEMS%%
}
